java/src/jni/h5Constants.c

```diff
@@ -1658,16 +1658,24 @@ Java_hdf_hdf5lib_HDF5Constants_H5FD_1MEM_1DEFAULT_1GHEAP_1SIZE(JNIEnv *env, jcla
 {
     return (hsize_t)(3 * (HADDR_MAX / (H5FD_MEM_NTYPES - 1)));
 }
+H5_GCC_CLANG_DIAG_OFF("sign-conversion")
 JNIEXPORT jlong JNICALL
 Java_hdf_hdf5lib_HDF5Constants_H5FD_1MEM_1DEFAULT_1LHEAP_1SIZE(JNIEnv *env, jclass cls)
 {
+    /* XXX: BADNESS! - This value cannot fit into a jlong!
```

Comment:
These are used only by one java test: testH5P_fapl_muti_defaults. I suggest we delete the test and all the java constants that start with "H5FD_MEM_DEFAULT_"

Reply:
If you do not want to remove the public constants then just set them to a value of -1.